dali/test/example_test.cc

```diff
@@ -0,0 +1,41 @@
+#include "dali_operator_test.h"
+
+namespace dali {
+
+namespace testing {
+
+struct InputData {
+  std::vector<int> data;
+  std::vector<size_t> shape;
+};
+
+std::vector<InputData> input_data;
+
+class MyOperatorTest : public DaliOperatorTest<int, float> {
+  std::vector<TensorAdapter<int>> SetInputs() const noexcept override {
+    std::vector<TensorAdapter<int>> ret;
+    for (auto in : input_data) {
+      ret.emplace_back(in.data, in.shape);
+    }
+    return ret;
+  }
+
+
+  OperatorsGraph SetOperators() const noexcept override {
+    return "MyOp";
+  };
+
+
+  bool Verify(TensorAdapter<float> output, TensorAdapter<float> anticipated_output) const noexcept override {
+    return false;
+  }
+};
+
+TEST_F(MyOperatorTest, MyOperatorTestCase) {
+  std::vector<TensorAdapter<float>> anticipated_outputs;
+  this->RunTest({{"arg1", true},
```

Comment:
@pribalta `OpSpec` defines whole operator - here we just supply arguments, which is good, IMO. It separates the definition of the pipeline from its configuration. If anything is a problem here, I'd say it's the fact that we're not addressing the problem of supplying argument inputs.

Reply:
The real problem here is that we still remain unable to have pipelines with more than one operator because the logic for building the pipeline is quite inconvenient. With the current approach, it will soon become awkward to parametrize more than one OpSpec. Hence, having an object acting as an adapter or builder for OpSpec could do the work just fine, while providing the insulation from OpSpec itself in our tests.

Reply:
Agree with @pribalta . Would you envision the `Arguments` here also as an adapter, that somehow maps provided arguments for proper operator inside graph? Something like
```
class ArgumentsAdapter {
  std::map<Operator, Arguments> map_;
};
```
This would solve the case when we want to have multiple ops in pipeline, so we need to provide them with arguments

Reply:
Would you kindly please stop calling everyghing "adapter"? It doesn't convey any meaning. If we need this extra class (which I don't think we do), then it should be "OperatorArgs". It could just as well be a typedef to the map.

Reply:
And what about the case with just one operator? We should make it as easy and short to write as possible Maybe we should have separate base class for single- and multi-operator tests? This would free the user who wants to test a single operator from specifying parameters that can only assume one value.

Reply:
Sure, let it be just `Arguments`. I used `Adapter` just to emphasize, that I intentionally used here well defined design pattern.

As for one operator case (which in fact will be most of the cases I think), I did similar thing in `OperatorsGraph`, i.e. a ctor:`OperatorsGraph(const char* operator_name);`
which can be used like:
`OperatorsGraph GenerateOperatorsGraph() const noexcept override {
    return "MyOp";
  };`
I would imagine, that this approach can be also applied for Arguments:
```
class Args {
  Args(std::map<std::string, TaggedUnion>);
  std::map<Operator, Arguments> map_;
};
```
(`TaggedUnion` is just an example, there are other ways to do it)